ocaml/gencert/gencert.ml

```diff
@@ -12,14 +12,24 @@
  * GNU Lesser General Public License for more details.
  *)
 
-(* Usage: gencert <path> *)
-
 let inventory = "/etc/xensource-inventory"
 
 module Lib = Gencertlib.Lib
 
 module D = Debug.Make (struct let name = "gencert" end)
 
+(* There are two types of certificates that can be generated, each one is
+   served depending on the SNI of the petition being received *)
+type sni = Default | Xapi_pool
```

Comment:
This looks good. I'm not sure that an extra type is worth it or whether binding names to strings would have been enough. The next step up would be a local module:

```
module SNI = struct
  type t = ..
  let of_string = function ..
end
```

This leads to nice names where this is used. Just an idea, though.